+Base/GitPanel.m

```diff
@@ -0,0 +1,164 @@
+            end
+            
+            obj.panel.Title = 'git branch';
+            obj.panel.Units = 'characters';
+            obj.panel.Position(4) = 2;
+            
+            filedetails = ' &lt;file(s)&gt; (--all)';
+            commitdetails = ' --author &lt;you&gt; --message &lt;description&gt;';
+            
+            obj.menu = uicontextmenu(f);
+            uimenu(obj.menu, 'Label',  '<html><font color="purple">git</font> fetch', 'Callback', @(s,e)obj.update)
+            uimenu(obj.menu, 'Label', ['<html><font color="purple">git</font> status<br>'...
+                                             '<font color="purple">git</font> ad&d' filedetails '<br>'...
+                                             '<font color="purple">git</font> commit' commitdetails '<br>'...
+                                             '<font color="purple">git</font> push origin <branch>'], 'Callback', @(s,e)(disp('Add files to a commit, then commit, and push to save changes.')));
+            uimenu(obj.menu, 'Label', ['<html><font color="purple">git</font> status<br>',...
+                                             '<font color="purple">git</font> pull'], 'Callback', @(s,e)(disp('Pull to stay up to date')))
+            
+            obj.panel.UIContextMenu = obj.menu; 
+            
+            obj.control = uicontrol('Parent', obj.panel, 'Style', 'checkbox', 'UIContextMenu', obj.menu, 'Units', 'characters', 'String', '', 'Tag', 'gitpanelcontrol');     % Text does not display HTML :(
+            obj.control.Position(1:2) = [-2.75 0];
+            obj.control.Position(3) = 200;
+            
+            if (obj.enableLeftClick)
+                obj.panel.ButtonDownFcn = @(s,e)(obj.update);
+                obj.control.Callback = @(s,e)obj.update;
+            end
+            
+            obj.update();
+        end
+        function update(obj)
+            obj.panel.Children.Enable = 'off';
+            obj.panel.HighlightColor = [.5 0 .5];   % Purple
+            obj.panel.Title = 'git fetch';
+            drawnow;
+            
+            try
+                git('fetch -q --all');
+                
+                obj.panel.Children.Tooltip = obj.tooltip();
+                obj.panel.Children.String = obj.info();
+                
+                obj.panel.Children.Enable = 'on';
+                obj.panel.HighlightColor = 'w';
+                obj.panel.Title = 'git branch';
+                drawnow;
+            catch err
+                obj.panel.Children.Enable = 'on';
+                obj.panel.Children.String = '<html>Could not <font color="purple">git</font> fetch.';
+                obj.panel.Children.Tooltip = 'Something terrible happened.';
+                drawnow;
+                
+                rethrow(err);
+            end
+        end
+        function thisbranch = thisbranch(obj) %#ok<MANU> It wasn't fetching properly when it was static.
+            thisbranch = split(git('branch', '-v'), '* ');
```

Comment:
Dangerous call to git!! See how it is done in [CommandCenter.m](https://github.com/mwalsh161/CommandCenter/blob/dev/CommandCenter.m#L148).

Primary issue is that you aren't explicitly forcing the git to evaluate in *this* repo's directory tree (if I remember, you might even need to be in the root of the repo).

This might also fix the static issue.

Reply:
See latest commit.